react/utils/metrics/impersonate.ts

```diff
@@ -0,0 +1,113 @@
+import type { Metric } from './metrics'
+import { sendMetric } from './metrics'
+
+type ImpersonatePerson = {
+  email: string
+  buy_org_id: string
+  cost_center_id: string
+  cost_center_name: string
+}
+
+type ImpersonateUser = ImpersonatePerson
+type ImpersonateTarget = ImpersonatePerson & { buy_org_name: string }
+
+type ImpersonateFieldsMetric = {
+  user: ImpersonateUser
+  target: ImpersonateTarget
+  date: string
+}
+
+type ImpersonateMetric = Metric & { fields: ImpersonateFieldsMetric }
+
+type ImpersonatePersonParams = {
+  costCenterId: string
+  costCenterName: string
+  organizationId: string
+  email: string
+}
+
+type ImpersonateUserParams = ImpersonatePersonParams
+type ImpersonateTargetParams = ImpersonatePersonParams & {
+  organizationName: string
+}
+
+export type ImpersonateMetricParams = {
+  account: string
+  target: ImpersonateTargetParams
+  user: ImpersonateUserParams
+}
+
+const buildMetric = (metricParams: ImpersonateMetricParams) => {
+  const { account, user, target } = metricParams
+
+  const metric = {
+    name: 'b2b-suite-buyerorg-data' as const,
+    account,
+    fields: {
+      user: {
+        email: user.email,
+        buy_org_id: user.organizationId,
```

Comment:
The correct is `buyer_org`, not `buy_org`, like you used in the metric's name a few lines above. Can you fix this and the other fields?